### ISSUE TITLE
Self override false positive

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#10409]
**Describe the bug**

```python
from typing_extensions import Self
from collections.abc import Sequence

class Foo:
    def bar(self, seq: Sequence[Self]):
        ...

class Foo2(Foo):
    def bar(self, seq: Sequence[Self]):
        ...
```

This fails to type check with override inconsistency error.  Error message is,

```
Method "bar" overrides class "Foo" in an incompatible manner
  Parameter 2 type mismatch: base parameter is type "Sequence[Foo]", override parameter is type "Sequence[Foo2]"
    "Sequence[Foo]" is not assignable to "Sequence[Foo2]"
      Type parameter "_T_co@Sequence" is covariant, but "Foo" is not a subtype of "Foo2"
        "Foo" is not assignable to "Foo2"  (reportIncompatibleMethodOverride)
```

**VS Code extension or command-line**
Using pyright playground I confirmed 1.1.398 type checks, 1.1.399 is when this starts failing.

**Comment by hmc-cs-mdrissi** [comment]
Thanks for the thread. I'll figure out how to adjust types then. Looks like best approach is make the class generic and follow CRTP pattern shown.
